src/command/expire/expire.c

```diff
@@ -502,9 +513,12 @@ removeExpiredArchive(InfoBackup *infoBackup)
                                 // Remove the entire directory if all archive is expired
                                 if (removeArchive)
                                 {
+                                  // execute the real expiration and deletion
+                                  // only if the dry-run mode is disabled
+                                  if ( ! cfgOptionBool(cfgOptDryRun) )
                                     storagePathRemoveP(
-                                        storageRepoWrite(), strNewFmt(STORAGE_REPO_ARCHIVE "/%s/%s", strPtr(archiveId),
-                                        strPtr(walPath)), .recurse = true);
+                                                       storageRepoWrite(), strNewFmt(STORAGE_REPO_ARCHIVE "/%s/%s", strPtr(archiveId),
+                                                                                     strPtr(walPath)), .recurse = true);
 
                                     archiveExpire.total++;
                                     archiveExpire.start = strDup(walPath);
```

Comment:
I cannot seem to comment on code that was not changed, but line 563:
```
// Remove archive log if it is not used in a backup
if (removeArchive)
{
    storageRemoveNP(
        storageRepoWrite(),
        strNewFmt(STORAGE_REPO_ARCHIVE "/%s/%s/%s",
            strPtr(archiveId), strPtr(walPath), strPtr(walSubPath)));
```
Should be wrapped so it looks like:
```
// Remove archive log if it is not used in a backup
if (removeArchive)
{
    // execute the real expiration and deletion only if the dry-run mode is disabled
    if (!cfgOptionBool(cfgOptDryRun))
    {
        storageRemoveNP(
            storageRepoWrite(),
            strNewFmt(STORAGE_REPO_ARCHIVE "/%s/%s/%s",
                strPtr(archiveId), strPtr(walPath), strPtr(walSubPath)));
    }
```

Reply:
thanks for pointing me to that